src/Microsoft.PowerShell.Commands.Management/commands/management/ContentCommandBase.cs

```diff
@@ -578,6 +578,15 @@ internal virtual bool CallShouldProcess(string path)
 
                 ErrorRecord pathNotFoundErrorRecord = null;
 
+                // Throw error if path is a directory.
+                if (System.IO.Directory.Exists(path))
+                {
+                    string errMsg = StringUtil.Format(SessionStateStrings.GetContainerContentException, path);
+                    ErrorRecord error = new ErrorRecord(new InvalidOperationException(errMsg), "GetContainerContentException", ErrorCategory.InvalidOperation, null);
+                    WriteError(error);
+                    return results;
```

Comment:
Below we call `SessionState.Path.GetResolvedPSPathFromPSPath()` which can return array of paths including containers.
I'd conside line 614 to the fix.

Reply:
The issue is provider specific (basically any provider that implements IContentCmdletProvider).  Looking in our code base, it seems the only public provider that's affected is the FileSystemProvider.  The fix should be in in [this method](https://github.com/PowerShell/PowerShell/blob/4e98f244fd3d4991d9b3005a856b6cd3e833870a/src/System.Management.Automation/namespaces/FileSystemProvider.cs#L6290).

Reply:
Then GetContentWriter should be fixed too.

Reply:
@SteveL-MSFT fix in that method will address only for `Get-Content` cmdlet. How about Set/Add-Content cmdlets ?

Reply:
@kvprasoon yes, we should fix those at the same time if possible as I imagine the change to be very similar.  I haven't looked at the code closely to see if there's a single point to make the change, but I suspect probably not.

Reply:
Single method what is identified is ResolvePaths() for *-Content cmdlets except for `Clear-Content`